internal/config/viper_store.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate go tool go.uber.org/mock/mockgen -destination=./mocks.go -package=config github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config ConfigStore
+
+package config
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+)
+
+type ViperConfigStore struct {
+	viper     viper.Viper
+	configDir string
+	fs        afero.Fs
+}
+
+// / ViperConfigStore specific methods
+func LoadViperStore(fs afero.Fs) (*ViperConfigStore, error) {
```

Comment:
(maybe I am missing something) I would have expected `viperStore` to implement the `store` interface, with `config` defining a `LoadStore` function that returns a generic `Store` interface. I’m wondering if you considered this design approach. The pros of this approach is that we will be able to easily remove viper with something else if needed int he future.

Reply:
Never mind my previous comment, I downloaded your PR and verified that you code is indeed doing what I was suggesting above. Great! 💯